### PR TITLE
Converts Buffers to strings by default or arrays if default is overridden

### PR DESCRIPTION
**Pull request by jcald1**
This PR solve problems with the serialization of Buffers.  For example, a Buffer representing the string 'ABC' (e.g., `Buffer.from('ABC', 'utf8')` ) is serialized as: `{ "0": 65, "1": 66, "2": 67 ]`.  This PR instead serializes it by default as 'ABC'.  The default can also be overridden to serialize it as an array: `[65, 66, 67]`.  The array is more useful than the current implementation because it can be easily converted into a string by the user.

The PR sets up an optional `options` parameter to be able to override defaults.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add a PR description that elaborates on what problem this solves and why it should be added here?

**Comment by jcald1** [comment]
@sindresorhus , done.

**Comment by sindresorhus** [comment]
I don't think this is a good default behavior. If you have something in a Buffer, there's a reason it's a Buffer and not a string. I also think this could be made more generally useful. Maybe a `serialize` and `deserialize` option where you can do whatever you want to prepare it. Example: https://github.com/sindresorhus/conf#serialize

File: test.js [hunk 1]
```diff
@@ -16,6 +19,44 @@ test('main', t => {
...
+	const serialized = serializeError(err, options);
+	t.is(serialized.child.buffer, 'ABC');
+});
+
+test('doesn\'t convert Buffers to strings when default is overridden', t => {
+	const options = {serializeError: {buffer: {toString: false}}};
+	const err = new Error('foo');
+	err.buffer = Buffer.from('ABC', 'utf8');
+	const serialized = serializeError(err, options);
+	t.true(_.isEqual(serialized.buffer, [...Buffer.from('ABC', 'utf8')]));
```
Comment (sindresorhus) [inline test.js]: There's already `t.deepEqual` for this.

File: package.json [hunk 2]
```diff
@@ -31,10 +31,13 @@
 		"deserialize"
 	],
 	"dependencies": {
-		"type-fest": "^0.12.0"
+		"buffer": "^5.5.0",
+		"type-fest": "^0.12.0",
+		"util": "^0.12.2"
```
Comment (sindresorhus) [inline package.json]: `buffer` and `util` are core modules. They should not be here.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "serialize-error",
-	"version": "6.0.0",
+	"version": "7.0.0",
```
Comment (sindresorhus) [inline package.json]: You should generally not bump the version in a PR. It's up to the maintainers to decide when and what version to use.

File: index.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 'use strict';
+const {Buffer} = require('buffer/');
```
Comment (sindresorhus) [inline index.js]: `Buffer` is a global. Prefer that one.

File: index.test-d.ts [hunk 5]
```diff
@@ -12,3 +12,4 @@ expectType<Error>(deserializeError({
 	name: 'name',
 	code: 'code'
 }));
+
```
Comment (sindresorhus) [inline index.test-d.ts]: Don't do unrelated changes.

File: package.json (hunk 2 above)
Comment (jcald1) [reply package.json]: @sindresorhus , wouldn't they be needed for frontend?

File: index.js (hunk 4 above)
Comment (jcald1) [reply index.js]: @sindresorhus , same comment as above (frontend)

File: package.json (hunk 2 above)
Comment (sindresorhus) [reply package.json]: No, it's up to browser user to polyfill those. Webpack does it automatically, I think.

**Comment by jcald1** [comment]
@sindresorhus , I've made the requested changed.  Converting to Buffer is not done by default and the options are added as part of the library initialization instead of each call to serializeError.

File: package.json (hunk 3 above)
Comment (jcald1) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (jcald1) [reply package.json]: fix

File: index.js (hunk 4 above)
Comment (jcald1) [reply index.js]: fixed

File: index.test-d.ts (hunk 5 above)
Comment (jcald1) [reply index.test-d.ts]: fixed

**Comment by sindresorhus** [comment]
> Don't do unrelated changes.

You're still doing a lot of unrelated changes. Look at the diff.
